clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/view/EcsInstanceProvider.java

```diff
@@ -0,0 +1,98 @@
+import com.netflix.spinnaker.clouddriver.ecs.cache.client.ContainerInstanceCacheClient;
+import com.netflix.spinnaker.clouddriver.ecs.cache.client.TaskCacheClient;
+import com.netflix.spinnaker.clouddriver.ecs.cache.model.ContainerInstance;
+import com.netflix.spinnaker.clouddriver.ecs.cache.model.Task;
+import com.netflix.spinnaker.clouddriver.ecs.model.EcsTask;
+import com.netflix.spinnaker.clouddriver.ecs.services.ContainerInformationService;
+import com.netflix.spinnaker.clouddriver.model.InstanceProvider;
+import org.apache.commons.lang.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.Map;
+
+@Component
+public class EcsInstanceProvider implements InstanceProvider<EcsTask> {
+
+  private final TaskCacheClient taskCacheClient;
+  private final ContainerInstanceCacheClient containerInstanceCacheClient;
+  private ContainerInformationService containerInformationService;
+
+  @Autowired
+  public EcsInstanceProvider(ContainerInformationService containerInformationService,
+                             TaskCacheClient taskCacheClient,
+                             ContainerInstanceCacheClient containerInstanceCacheClient) {
+    this.containerInformationService = containerInformationService;
+    this.taskCacheClient = taskCacheClient;
+    this.containerInstanceCacheClient = containerInstanceCacheClient;
+  }
+
+  @Override
+  public String getCloudProvider() {
+    return EcsCloudProvider.ID;
+  }
+
+  @Override
+  public EcsTask getInstance(String account, String region, String id) {
+    if (!isValidId(id, region))
+      return null;
+
+    EcsTask ecsInstance = null;
+
+    String key = Keys.getTaskKey(account, region, id);
+    Task task = taskCacheClient.get(key);
+    if (task == null) {
+      return null;
+    }
+
+    key = Keys.getContainerInstanceKey(account, region, task.getContainerInstanceArn());
+    ContainerInstance containerInstance = containerInstanceCacheClient.get(key);
+
+    if (containerInstance != null) {
+      String serviceName = StringUtils.substringAfter(task.getGroup(), "service:");
+      Long launchTime = task.getStartedAt();
+
+      List<Map<String, String>> healthStatus = containerInformationService.getHealthStatus(id, serviceName, account, region);
+      String address = containerInformationService.getTaskPrivateAddress(account, region, task);
+
+      ecsInstance = new EcsTask(id, launchTime, task.getLastStatus(), task.getDesiredStatus(), containerInstance.getAvailabilityZone(), healthStatus, address);
+    }
+
+    return ecsInstance;
+  }
+
+  @Override
+  public String getConsoleOutput(String account, String region, String id) {
+    return null;
+  }
+
+  private boolean isValidId(String id, String region) {
+    String id_regex = "[\\da-f]{8}-[\\da-f]{4}-[\\da-f]{4}-[\\da-f]{4}-[\\da-f]{12}";
+    String id_only = String.format("^%s$", id_regex);
```

Comment:
Please camelCase the things.

Reply:
Done.

Think I was writing Python that day and it carried over :)